### PR TITLE
chore:Update docs for Android SDK

### PR DESCRIPTION
**Pull request by typotter**
""never a chore to update the docs" - Wayne Gretzky" - Tyler Potter

### CODE REVIEW

**Review by felipecsl — APPROVED** [review]
Nice docs!

File: docs/sdks/client-sdks/android.md [hunk 1]
```diff
@@ -13,26 +13,83 @@ Eppo's open source Android SDK can be used for both feature flagging and experim
...
+
 During initialization, the SDK sends an API request to a CDN to retrieve the most recent experiment configurations from Eppo, 
 such as variation values and traffic allocation. The SDK stores these configurations in memory so that assignments are effectively instant. 
 For more information, see the [architecture overview](/sdks/architecture) page. 
 
 This SDK also leverages cached configurations from previous fetches. During initialization, if a previously-cached configuration 
 is successfully loaded, it will complete initialization with that configuration. Updates will take effect once the fetch from CDN completes.
-When initialization completes from either source, it will call the optionally-provided `InitializationCallback`.
+
+The initialization methods, `buildAndInit` and `buildAndInitAsync` return or resolve to an `EppoClient` instance with a loaded configuration.
```
Comment (aarsilv) [inline docs/sdks/client-sdks/android.md:44]: ❤️

File: docs/sdks/client-sdks/android.md [hunk 2]
```diff
@@ -13,26 +13,83 @@ Eppo's open source Android SDK can be used for both feature flagging and experim
...
+    interface MyAppConfig {
+        String getEppoConfig();
+    }
+
+    // Initialize the Eppo Client with the async result of the initial client configuration.
+
+    // 1. Fetch your app's initial configuration
+    CompletableFuture<MyAppConfig> initialconfigFuture = ...;
+
+    // 2. Transoform the result into just the Eppo Configuration string (byte array).
```
Comment (aarsilv) [inline docs/sdks/client-sdks/android.md:61]: typo

File: docs/sdks/client-sdks/android.md [hunk 3]
```diff
@@ -13,26 +13,83 @@ Eppo's open source Android SDK can be used for both feature flagging and experim
...
+    CompletableFuture<byte[]> eppoConfigBytes = initialconfigFuture.thenApply(
+        ic-> ic.getEppoConfig().getBytes());
+
+    // 3. Build the client with the asycn initial config
```
Comment (aarsilv) [inline docs/sdks/client-sdks/android.md:65]: typo here too!

**Review by aarsilv — APPROVED** [review]
👏

**Comment by aarsilv** [comment]
Love the PR description quote 😆

File: docs/sdks/client-sdks/android.md (hunk 3 above)
Comment (typotter) [reply docs/sdks/client-sdks/android.md]: oof

**Pull request merged**